crates/holochain_p2p/tests/tests/unresponsive_url_pruning.rs

```diff
@@ -75,37 +75,75 @@ async fn urls_are_pruned_when_updated_agent_info_available() {
         db_peer_meta,
     } = TestCase::spawn().await;
 
+    let earliest_timestamp = Timestamp::from_micros(Timestamp::now().as_micros() - 10_000_000); // 10s ago
+
     // Insert an unresponsive URL into peer meta store.
     let unresponsive_url = Url::from_str("ws://under.water:80").unwrap();
     let expiry = Timestamp::from_micros(Timestamp::now().as_micros() + 10_000_000); // 10 s from now
     peer_meta_store
         .set_unresponsive(unresponsive_url.clone(), expiry, Timestamp::now())
         .await
         .unwrap();
-    peer_meta_store
-        .set_unresponsive(unresponsive_url.clone(), expiry, Timestamp::now())
+    let maybe_unresponsive_url = peer_meta_store
+        .get_unresponsive(unresponsive_url.clone())
         .await
         .unwrap();
+    assert!(maybe_unresponsive_url.is_some());
+
     // Mark another URL unresponsive, for which there won't be an updated agent info.
     let other_unresponsive_url = Url::from_str("ws://under.earth:80").unwrap();
     peer_meta_store
         .set_unresponsive(other_unresponsive_url.clone(), expiry, Timestamp::now())
         .await
         .unwrap();
-
-    let maybe_unresponsive_url = peer_meta_store
-        .get_unresponsive(unresponsive_url.clone())
-        .await
-        .unwrap();
-    assert!(maybe_unresponsive_url.is_some());
     let maybe_other_unresponsive_url = peer_meta_store
         .get_unresponsive(other_unresponsive_url.clone())
         .await
         .unwrap();
     assert!(maybe_other_unresponsive_url.is_some());
 
+    // Insert an agent info into peer store which was created before the first unresponsive peer meta store value
     let agent = lair_client.new_sign_keypair_random().await.unwrap();
-    let local_agent = HolochainP2pLocalAgent::new(agent.clone(), DhtArc::FULL, 1, lair_client);
+    let local_agent =
+        HolochainP2pLocalAgent::new(agent.clone(), DhtArc::FULL, 1, lair_client.clone());
+    let updated_agent_info = AgentInfoSigned::sign(
+        &local_agent,
+        AgentInfo {
+            agent: AgentId(Id(Bytes::from_static(&[0xaa; 32]))),
+            created_at: earliest_timestamp,
+            expires_at: Timestamp::from_micros(Timestamp::now().as_micros() + 10_000_000),
+            is_tombstone: false,
+            space: space_id.clone(),
+            storage_arc: DhtArc::FULL,
+            url: Some(unresponsive_url.clone()),
+        },
+    )
+    .await
+    .unwrap();
+    p2p.test_kitsune()
+        .space(space_id.clone())
+        .await
+        .unwrap()
+        .peer_store()
+        .insert(vec![updated_agent_info])
+        .await
+        .unwrap();
+
+    // Wait for at least one pruning.
+    tokio::time::sleep(Duration::from_millis(500)).await;
+
+    // URL should not have been removed from the store.
```

Comment:
```suggestion
    // The peer should still be marked unresponsive because it was marked unresponsive in the peer meta store later than the creation time of the newest agent info of that peer.
```